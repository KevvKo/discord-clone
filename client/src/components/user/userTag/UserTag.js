function UserTag() {

    const userName = "unosxmopsus";
    const userTag = "#3432";

    return (
    <div className="user-tag flex-fill">
        <div>{userName}</div>
        <div>{userTag}</div>
    </div>
    );
}

export default UserTag;
