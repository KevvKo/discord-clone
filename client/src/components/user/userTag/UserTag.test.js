import React from 'react';
import UserTag from './UserTag';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { act } from 'react-dom/test-utils';
const mockStore = configureStore([]);
 
describe('UserAvatar', () => {
    let store; 
    let component;
    beforeEach(() => {
        store = mockStore({
            user: {
                id: 1,
                name: 'xyz name',
                email: 'xyz@mail.com',
                phoneNumber: '12345678',
                status: true,
                active: true 

            }
        });

        component = renderer.create( 
            <Provider store={store}>
                <UserTag />
            </Provider>
        );
    });
    it('should render with given state from redux store', () => {
        expect(component.toJSON()).toMatchSnapshot();
        act( () => {
            expect(
                component.root.findAllByType('div')[1].props.children
            ).toEqual('xyz name');
            expect(
                component.root.findAllByType('div')[2].props.children
            ).toEqual(1);
        });
    });
});
