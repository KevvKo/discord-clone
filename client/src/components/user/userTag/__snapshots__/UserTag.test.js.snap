// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAvatar should render with given state from redux store 1`] = `
<div
  className="user-tag flex-fill"
  data-testid="user-tag"
>
  <div>
    xyz name
  </div>
  <div>
    #
    1
  </div>
</div>
`;
